### ISSUE TITLE
cleanup py3compat

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#11985]
A number of utilities function in `IPython/utils/py3compat.py` are no-op, or could be simplified. 

For example the following two variables :

```
string_types = (str,)
unicode_type = str
```

And following functions:

```
str_to_unicode = no_code
unicode_to_str = no_code
cast_bytes_py2 = no_code
cast_unicode_py2 = no_code
buffer_to_bytes_py2 = no_code
```

Are often use in a single place or do nothing. Their usage should be replaced / removed.

**Comment by Bhanu-mbvg** [comment]
Can I take up this issue?

**Comment by Carreau** [comment]
closed by #12009
